src/main/java/org/mockito/internal/verification/MockAwareVerificationMode.java

```diff
@@ -4,21 +4,39 @@
  */
 package org.mockito.internal.verification;
 
-import org.mockito.internal.verification.api.VerificationData;
+import java.util.Set;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.verification.VerificationData;
+import org.mockito.listeners.VerificationListener;
 import org.mockito.verification.VerificationMode;
+import org.mockito.verification.VerificationSucceededEvent;
 
 public class MockAwareVerificationMode implements VerificationMode {
 
     private final Object mock;
     private final VerificationMode mode;
+    private final Set<VerificationListener> listeners;
 
-    public MockAwareVerificationMode(Object mock, VerificationMode mode) {
+    public MockAwareVerificationMode(Object mock, VerificationMode mode, Set<VerificationListener> listeners) {
         this.mock = mock;
         this.mode = mode;
+        this.listeners = listeners;
     }
 
     public void verify(VerificationData data) {
-        mode.verify(data);
+        try {
+            mode.verify(data);
+
+            for (VerificationListener listener : listeners) {
+                listener.onVerificationSucceeded(new VerificationSucceededEvent(mock, mode, data));
+            }
+        } catch (MockitoAssertionError e) {
+            for (VerificationListener listener : listeners) {
+                listener.onVerificationException(mock, mode, e);
```

Comment:
@LiamClark, nice catch!

Let's catch both, RuntimeException and Error types. This way, you can rethrow while keeping the implementation robust enough to support both cases.